### ISSUE TITLE
charm_lifecycle feature: upgrade_charm

### ISSUE THREAD

**Issue by auria** [openstack-charmers/zaza#352]
Hi,

As part of the steps to test a new fix works as it should, it would be of great help a Juju helper to upgrade a charm from the charm store latest version to the local fix. I've seen [1] (`zaza.model.get_application(appname)`) and [2] (`juju.application.upgrade_charm`).

Do you recommend to use both functions to run this kind of testing? Do you think `zaza.utilities` should include an upgrade_charm façade?

Thank you,
-Alvaro.

1. https://github.com/openstack-charmers/zaza/blob/master/zaza/model.py#L419
2. https://github.com/juju/python-libjuju/blob/master/juju/application.py#L448

**Comment by ChrisMacNaughton** [comment]
I don't think that there's a huge benefit to Zaza wrapping libjuju's [`application.upgrade_charm`](https://github.com/juju/python-libjuju/blob/master/juju/application.py#L448) as that is available directly as a dependency of Zaza.
